Source/CombatExtended/CombatExtended/Things/Smoke.cs

```diff
@@ -21,7 +22,9 @@ public override void ExposeData()
 
         private bool CanMoveTo(IntVec3 pos)
         {
-            return !pos.Filled(Map) || (pos.GetDoor(Map)?.Open ?? false) || pos.GetFirstThing<Building_Vent>(Map) != null;
+            return !pos.Filled(Map) 
+                || (pos.GetDoor(Map)?.Open ?? false) 
+                || (pos.GetFirstThing<Building_Vent>(Map) is Building_Vent vent && Traverse.Create(vent).Field("flickableComp").GetValue<CompFlickable>().SwitchIsOn);
```

Comment:
Why are you using reflection for this?

Reply:
Because Ludeon is silly and made Building_Vent.flickableComp a private field. Is there a simpler way to get at it?

Reply:
.GetComponent<>()?